.github/workflows/lint.yml

```diff
@@ -1,32 +1,63 @@
-name: Lint
+name: Lint and test
 on:
   push:
-    branches: [ main, develop ]
+    branches: [main, develop]
   pull_request:
-    branches: [ main, develop ]
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+
 jobs:
   lint:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         python-version: ["3.12"]
     steps:
-    - uses: actions/checkout@v4
-    - name: Install uv
-      uses: astral-sh/setup-uv@v3
-      with:
-        version: "latest"
-    - name: Set up Python ${{ matrix.python-version }}
-      run: uv python install ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        cd notebook_utils
-        uv sync --inexact --group dev
-    - name: Run linting
-      run: |
-        cd notebook_utils
-        uv run ruff check ..
-    - name: Check formatting
-      run: |
-        cd notebook_utils
-        uv run ruff format --check ..
\ No newline at end of file
+      - uses: actions/checkout@v5
+      - name: Install uv
+        uses: astral-sh/setup-uv@v7
+        with:
+          version: "latest"
+      - name: Set up Python ${{ matrix.python-version }}
+        run: uv python install ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          cd notebook_utils
+          uv sync --inexact --group dev
+      - name: Run linting
+        run: |
+          cd notebook_utils
+          uv run ruff check ..
+      - name: Check formatting
+        run: |
+          cd notebook_utils
+          uv run ruff format --check ..
+
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.12", "3.13"]
+    steps:
+      - uses: actions/checkout@v5
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Build Docker image
+        run: |
+          echo "Building Docker image..."
+          docker build -t spark-notebook-test . || {
+            echo "Docker build failed. This might be due to network issues or base image availability."
+            echo "Checking if base image exists..."
+            docker pull ghcr.io/berdatalakehouse/spark_notebook_base || echo "Base image pull failed"
+            exit 1
+          }
+
+      - name: Run container test
+        run: |
+          echo "Running test inside container..."
+          # slightly nasty hack
+          chmod 777 ./notebook_utils
+          docker run --rm -e NB_USER=runner -v ./notebook_utils:/tmp/notebook_utils spark-notebook-test:latest /bin/bash /tmp/notebook_utils/run_tests.sh
```

Comment:
mount the notebook_utils dir directly in the docker image and run the tests from there